compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -495,6 +496,29 @@ class Semantic {
           Result(value2, errors)
       }
     }
+
+    def accessLocal(sym: Symbol, default: Result, klass: ClassSymbol, source: Tree): Contextual[Result] =
+      sym.defTree match {
+        case vdef: ValDef =>
+          // resolve this for local variable
+          val enclosingClass = sym.owner.enclosingClass.asClass
+          val thisValue2 = resolveThis(enclosingClass, value, klass, source)
+          thisValue2 match {
+            case Hot => Result(Hot, Errors.empty)
+            
+            case Cold => 
+              val error = AccessCold(sym, source, trace.toVector)
+              Result(Hot, error :: Nil)
+            
+            case addr: Addr => 
+              val res = eval(vdef.rhs, addr, klass)
+              if res.value.promote("Try promote", source).isEmpty then Result(Hot, Errors.empty) else res
```

Comment:
Why we call `promote` here?

Reply:
It's tests/pos/i8892.scala. If we don't promote here, dotty will crash on this test.

Reply:
Thanks for the info. The crash seems to imply there is some bug elsewhere, it's difficult to justify the promotion here.

Could you try to debug the test and see why it crashes?

Reply:
OK, I'll take a closer look at it.

Reply:
@liufengyun It seems that we are selecting a non-exist field `1_<outer>` from `Zero` in `val j31`. I think we are meant to call `Zero.++`, but "++" is transparent inlined def, so it somehow got translated to `1_<outer>`

Reply:
@EnzeXing You can try the following patch: https://gist.github.com/liufengyun/de6198e4a8a28e090c41e62c8af65ed6

Reply:
With this patch, this test is resolved, but now we would fail on tests/init/special/i12128/Test_3.scala since dotty can't identify the object `MacroCompat`

Reply:
It works without any problem for me (including the test above). Could you please debug?

Reply:
nvm, this has been resolved.